File: java/app/src/main/java/com/pgssoft/mvvm/views/adapters/RatesAdapter.java
package com.pgssoft.mvvm.views.adapters;

import android.content.Context;

import com.pgssoft.mvvm.BR;
import com.pgssoft.mvvm.R;
import com.pgssoft.mvvm.model.database.Rate;
import com.pgssoft.mvvm.viewmodels.RateViewModel;
import com.pgssoft.mvvm.views.adapters.interfaces.handlers.RateItemHandler;
import com.pgssoft.mvvm.views.adapters.interfaces.providers.BaseItemProvider;

import java.util.List;

/**
 * Created by bstokrocki on 07.02.2017.
 */

public class RatesAdapter extends BindingRecyclerAdapter<Rate, RateViewModel> {
    private RateItemHandler rateItemHandler;

    public RatesAdapter(Context context, BaseItemProvider<Rate> itemProvider, RateItemHandler rateItemHandler) {
        super(context, itemProvider.getItems(), BR.viewModel, R.layout.list_item_rate);
        this.rateItemHandler = rateItemHandler;
    }


    public RateViewModel createViewModel() {
        return new RateViewModel(rateItemHandler);
    }
}

File: kotlin/app/src/main/java/com/pgssoft/kotlinplayground/view/adapter/RatesAdapter.kt
package com.pgssoft.kotlinplayground.view.adapter

import android.content.Context
import com.pgssoft.kotlinplayground.BR
import com.pgssoft.kotlinplayground.R
import com.pgssoft.kotlinplayground.model.db.Rate
import com.pgssoft.kotlinplayground.view.adapter.handlers.RateItemHandler
import com.pgssoft.kotlinplayground.view.adapter.providers.BaseItemProvider
import com.pgssoft.kotlinplayground.viewmodel.RateViewModel
import com.pgssoft.mvvm.views.adapters.BindingRecyclerAdapter

/**
 * Created by bstokrocki on 07.02.2017.
 */

class RatesAdapter(context: Context, itemProvider: BaseItemProvider<Rate>, private val rateItemHandler: RateItemHandler)
    : BindingRecyclerAdapter<Rate, RateViewModel>(context, itemProvider.items, BR.viewModel, R.layout.list_item_rate) {

    override fun createViewModel(): RateViewModel {
        return RateViewModel(rateItemHandler)
    }

    fun filter(itemProvider: BaseItemProvider<Rate>) {
        dataItems = itemProvider.items
        notifyDataSetChanged()
    }
}
